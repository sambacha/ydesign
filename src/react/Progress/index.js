import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { themeGet } from '@styled-system/theme-get';
import { transparentize } from 'polished';

import theme from '../theme';

const StyledProgress = styled.progress.attrs(props => ({
  primarycolor: themeGet('colors.primary', '#000')(props),
}))`
  & {
    appearance: none;
    display: inline-block;
    overflow: hidden;
    height: 8px;
    border-radius: 8px;
  }
  &::-webkit-progress-bar {
    background-color: ${props => transparentize('0.75', props.primarycolor)};
  }
  &::-webkit-progress-value {
    background-color: ${props => props.primarycolor};
    border-radius: 8px;
  }
`;

const Progress = React.forwardRef((props, ref) => (
  <StyledProgress ref={ref} {...props} />
));

StyledProgress.defaultProps = {
  theme,
};

// Progress.defaultProps = {
//   theme,
// };

Progress.propTypes = {
  theme: PropTypes.object,
};

Progress.displayName = 'Progress';

export default Progress;
