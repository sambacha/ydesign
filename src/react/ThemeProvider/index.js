import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider as StyledThemeProvider } from 'styled-components';
import theme from '../theme';

const ThemeProvider = ({ children, theme, ...rest }) => {
  return (
    <StyledThemeProvider theme={theme} {...rest}>
      {children}
    </StyledThemeProvider>
  );
};

ThemeProvider.propTypes = {
  theme: PropTypes.object,
};

ThemeProvider.defaultProps = {
  theme,
};

ThemeProvider.displayName = 'ThemeProvider';

export default ThemeProvider;
