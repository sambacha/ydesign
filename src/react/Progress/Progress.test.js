import Progress from './';
import ThemeProvider from '../ThemeProvider';
import theme from '../theme';

describe('Progress component sanity', () => {
  it('has name', () => {
    expect(Progress.displayName).toBe('Progress');
  });

  it('matches default snapshot', () => {
    const component = render(<Progress />);
    expect(component).toMatchSnapshot();
  });

  it('matches themed snapshot', () => {
    const component = render(
      <ThemeProvider theme={theme}>
        <Progress />
      </ThemeProvider>
    );
    expect(component).toMatchSnapshot();
  });
});
