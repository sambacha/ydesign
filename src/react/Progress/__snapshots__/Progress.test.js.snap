// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress component sanity matches default snapshot 1`] = `
.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  overflow: hidden;
  height: 8px;
  border-radius: 8px;
}

.c0::-webkit-progress-bar {
  background-color: rgba(78,63,206,0.25);
}

.c0::-webkit-progress-value {
  background-color: #4E3FCE;
  border-radius: 8px;
}

<progress
  class="c0"
/>
`;

exports[`Progress component sanity matches themed snapshot 1`] = `
.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  overflow: hidden;
  height: 8px;
  border-radius: 8px;
}

.c0::-webkit-progress-bar {
  background-color: rgba(78,63,206,0.25);
}

.c0::-webkit-progress-value {
  background-color: #4E3FCE;
  border-radius: 8px;
}

<progress
  class="c0"
/>
`;
